### DISCUSSION TITLE
is langium  support validate in-memory document

### DISCUSSION THREAD

**Discussion by zrwsmd** [eclipse-langium/langium#1521] [Q&A] [answered]
follow the #984,now i can refer outer model correctly,similar to oridinary st file,validation is also necessary,is langium now can support validate in-memory document  as well(other st file refer in-memory document )
![image](https://github.com/eclipse-langium/langium/assets/26995040/857a2c35-6862-42fa-904f-e7b500ee693a)

**Comment by msujew** [comment, marked as answer]
@zrwsmd What is supposed to be validated? By default, missing references are validated, but nothing else. In case you want type system based validations, you need to implement them yourself.

Note that in-memory documents that haven't been parsed from text cannot be validated, as the validator requires text-offsets to work correctly.

**Reply by zrwsmd** [reply]
yes,i want to type system based validations ,for example, `c1.ccc` is string type,so above screenshot ,INT can't be transform into STRING

**Reply by msujew** [reply]
See [langium-lox](https://github.com/TypeFox/langium-lox). It's an example language with a type system implementation.
